pkg/controller/perconaservermongodbbackup/perconaservermongodbbackup_controller.go

```diff
@@ -0,0 +1,172 @@
+	err = c.Watch(&source.Kind{Type: &psmdbv1alpha1.PerconaServerMongoDBBackup{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// TODO(user): Modify this to be the types you create that are owned by the primary resource
+	// Watch for changes to secondary resource Pods and requeue the owner PerconaServerMongoDBBackup
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &psmdbv1alpha1.PerconaServerMongoDBBackup{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcilePerconaServerMongoDBBackup{}
+
+// ReconcilePerconaServerMongoDBBackup reconciles a PerconaServerMongoDBBackup object
+type ReconcilePerconaServerMongoDBBackup struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a PerconaServerMongoDBBackup object and makes changes based on the state read
+// and what is in the PerconaServerMongoDBBackup.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
+// a Pod as an example
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcilePerconaServerMongoDBBackup) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling PerconaServerMongoDBBackup")
+	rr := reconcile.Result{
+		RequeueAfter: time.Second * 5,
+	}
+	// Fetch the PerconaServerMongoDBBackup instance
+	instance := &psmdbv1alpha1.PerconaServerMongoDBBackup{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			log.Info("Backup get:" + err.Error())
+			return rr, nil
+		}
+		// Error reading the object - requeue the request.
+		log.Info("Backup get:" + err.Error())
+		return rr, err
+	}
+
+	err = r.reconcileBC(instance)
+	if err != nil {
+		log.Error(err, "BackupHandler: ")
+		return rr, err
+	}
+
+	err = r.updateStatus(instance)
+	if err != nil {
+		log.Info("Backup status:" + err.Error())
+		return rr, err
+	}
+
+	return rr, nil
+}
+
+func (r *ReconcilePerconaServerMongoDBBackup) reconcileBC(cr *psmdbv1alpha1.PerconaServerMongoDBBackup) error {
+	if len(cr.Name) == 0 || len(cr.Spec.StorageName) == 0 || len(cr.Spec.PSMDBCluster) == 0 {
+		return fmt.Errorf("not enough data")
```

Comment:
1. Confusing error message. What data is not enough? Where this data should be? Check all required fields from cr separately and return relevant errors.
2. It's better to follow existing code. Either put it in separate method of `psmdbv1alpha1.PerconaServerMongoDBBackup` type or, if there are few checks, do it in the `Reconcile` right after the instance was fetched